app/models/periodic_expense.rb

```diff
@@ -0,0 +1,50 @@
+class PeriodicExpense < ActiveRecord::Base
+  PERIODS = { 'weekly' => 1.week, 'monthly' => 1.month }
+
+  belongs_to :user
+
+  validates :name, :user, :amount, :period, :start_date, presence: true
+  validates :period, inclusion: { in: PERIODS.keys }
+  validate :end_date_is_after_start_date
+  validate :last_paid_at_is_between_start_and_end_dates
+
+  def period_as_time
+    PERIODS[period]
+  end
+
+  def ready_to_pay?
+    !next_pay.future?
```

Comment:
It's also a little easier to read "next pay at future?"

Actually, it's in the guides:

> - Name date columns with _on suffixes.
> - Name datetime columns with _at suffixes.

Following this (although it says it's for columns), the suffix should `_on`, and the same applies to `last_paid_at`.

What about `start_date` and `end_date`?